src/shared/component/CommentBox/hook.ts

```diff
@@ -0,0 +1,63 @@
+"use client";
+
+import { useEditCommentMutation } from "@/app/group/[groupId]/solved-detail/[id]/query";
+import { CommentsContext } from "@/view/group/solved-detail/CommentSection/provider";
+import { type KeyboardEvent, useContext } from "react";
+import { flushSync } from "react-dom";
+import { type SubmitHandler, useForm } from "react-hook-form";
+
+type EditForm = {
+  input: string;
+};
+export const useEditForm = (commentId: number, defaultValue: string) => {
+  const { editingItem, handleEditItem, handleReset, solutionId } =
+    useContext(CommentsContext);
+
+  const { register, setValue, setFocus, handleSubmit } = useForm<EditForm>({
+    defaultValues: {
+      input: defaultValue,
+    },
+  });
+
+  const { mutate: editMutate } = useEditCommentMutation(solutionId, commentId);
+
+  const isEditing = editingItem === commentId;
+
+  const handleEditBtnClick = () => {
+    flushSync(() => {
+      handleEditItem(commentId);
+    });
+
+    setFocus("input");
```

Comment:
```suggestion
    if (!isEditing) {
      setFocus("input");
    }
```

편집 버튼 클릭 시 항상 입력 필드에 포커스가 설정되지만, 이미 포커스가 설정된 상태일 때 중복 호출이 발생할 수 있을 것 같습니다!
해당 분기를 추가해야하지 않을까요? 제가 로직을 잘못 이해한거면 설명 부탁드립니다!

Reply:
"편집 버튼 클릭 시 입력 필드에 포커스가 되도록" 하는 로직입니다 ! setFoucs를 호출하지 않고 현재 editing 중인 아이템만 수정한다면 당연히 포커스가 되지 않습니다